openmc/mgxs/mgxs.py

```diff
@@ -4528,41 +4776,45 @@ def xs_tally(self):
         return self._xs_tally
 
 
-class NuFissionMatrixXS(MatrixMGXS):
-    r"""A fission production matrix multi-group cross section.
+class ScatterProbabilityMatrix(MatrixMGXS):
```

Comment:
I left this class intact since it could be useful for educational or illustrative purposes. I'm refactoring it by removing support for higher order Legendre moments so that it doesn't encapsulate so much redundant code, and I really can't think of a use case for higher order moments in this case.